shell/platform/fuchsia/flutter/accessibility_bridge.cc

```diff
@@ -309,15 +320,62 @@ void AccessibilityBridge::AddSemanticsNodeUpdate(
     PrintNodeSizeError(nodes.back().node_id());
   }
 
+  // Even if the SemanticsNodeUpdates does not contain the root node, we still
+  // need to generate an update for the root node in case the view pixel ratio
+  // has changed.
+  std::vector<fuchsia::accessibility::semantics::Node> root_update;
+  root_update.push_back(GetRootNodeUpdate());
+
   PruneUnreachableNodes();
   UpdateScreenRects();
 
-  tree_ptr_->UpdateSemanticNodes(std::move(nodes));
+  tree_ptr_->UpdateSemanticNodes(std::move(root_update));
+  if (!nodes.empty())
+    tree_ptr_->UpdateSemanticNodes(std::move(nodes));
   // TODO(dnfield): Implement the callback here
   // https://bugs.fuchsia.dev/p/fuchsia/issues/detail?id=35718.
   tree_ptr_->CommitUpdates([]() {});
 }
 
+fuchsia::accessibility::semantics::Node
+AccessibilityBridge::GetRootNodeUpdate() {
+  fuchsia::accessibility::semantics::Node root_fuchsia_node;
+  std::vector<uint32_t> child_ids;
+  size_t this_node_size = sizeof(fuchsia::accessibility::semantics::Node);
+  for (int32_t flutter_child_id :
+       root_flutter_semantics_node_.childrenInTraversalOrder) {
+    child_ids.push_back(FlutterIdToFuchsiaId(flutter_child_id));
+  }
+  // Applies the inverse view pixel ratio transformation to the root node.
+  float view_pixel_ratio = delegate_.GetViewPixelRatio();
+  float inverse_view_pixel_ratio = 1.f / view_pixel_ratio;
+  SkM44 inverse_view_pixel_ratio_transform;
+  inverse_view_pixel_ratio_transform.setScale(inverse_view_pixel_ratio,
+                                              inverse_view_pixel_ratio, 1.f);
+
+  SkM44 result = root_flutter_semantics_node_.transform *
+                 inverse_view_pixel_ratio_transform;
+  nodes_[root_flutter_semantics_node_.id] = {
+      .id = root_flutter_semantics_node_.id,
+      .flags = root_flutter_semantics_node_.flags,
+      .rect = root_flutter_semantics_node_.rect,
+      .transform = result,
+      .children_in_hit_test_order =
+          root_flutter_semantics_node_.childrenInHitTestOrder,
+  };
```

Comment:
Isn't this replacing whatever was on the root node previously? Don't we _only_ want to update the transform here?

Also, what if the root node came in with a transform already, that we're now no longer respecting?

Reply:
line356 does `root_flutter_semantics_node_.transform * inverse_view_pixel_ratio_transform;`

This should respect the origin transform.
The reason why i don't update the original transform directly is that we need to memorize the original transform in case the root node may not be in the next update(where root_flutter_semantics_node_ will be the previous cached value)